Automattic/wp-calypso
# Add `restricted_tld` option to the request list.

#### Proposed Changes

This PR adds the `allowed_tld` site option to the list of the requested site options, which in turn stores it in the state tree. The site option is introduced in https://github.com/Automattic/jetpack/pull/28087 and code-D96551.

#### Testing Instructions

1. Follow the intruction of https://github.com/Automattic/jetpack/pull/28087#issuecomment-1365874643 to deploy the endpoint PR to your sandbox.
2. Apply code-D96551. I'd recommend applying it with `--nobranch` and `--nocommit` to make it simpler.
3. Create a site through `/setup/link-in-bio-tld` flow.
4. Open up the browser console and inspect the state by `getState().sites.items`. `restricted_tld` should appear:
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/1842898/209693774-75732b5b-0345-4494-93b8-ffdd33278d91.png">
5. Create another site through some other flows and repeat the 4th step. The `restricted_tld` field should be `false`.



#### Pre-merge Checklist





Related to Automattic/martech#1320
